vscode/syntaxes/toit.tmLanguage.yaml

```diff
@@ -459,10 +459,12 @@ repository:
     #  - single capitalized character -> a type: 'class A'.
     #  - single capitalized character followed by a number -> also a type: `class C1`.
     #  - capitalized identifier with at least on lower-case character: a type.
-    - name: entity.name.type.toit
-      match: \b(?<!-)_?[A-Z][0-9]*(?<!-)\b[?]?
+
+    # Match the longer type first, so that we match something like `X-Of`.
     - name: entity.name.type.toit
       match: \b(?<!-)_?[A-Z][A-Z_-]*[a-z][\w-]*(?<!-)\b[?]?
+    - name: entity.name.type.toit
+      match: \b(?<!-)_?[A-Z][0-9]*(?<!-)\b[?]?
```

Comment:
Do we have special handling for prefix underscores here?

Reply:
Not sure why we have it. (Maybe still old privacy syntax).
That said, I think I would expect `_A` to be a type.
In the end it's just an heuristic.

That said: it doesn't cover the `A_` case, which should also be a type.
Added that.
done.